src/main/java/com/glencoesoftware/omero/zarr/OmeroAmazonS3ClientFactory.java

```diff
@@ -80,16 +106,47 @@ private String getBucketFromUri(URI uri) {
         return path.substring(0, path.indexOf("/"));
     }
 
+    /**
+     * Retrieves the region from a given URI.
+     * @param uri The URI to handle
+     * @return The region
+     */
     private String getRegionFromUri(URI uri) {
         String host = uri.getHost();
-        return host.split("\\.")[1];
+        if (host.contains("amazonaws.com")) {
+            String[] values = host.split("\\.");
+            if (values.length == 3) {
+                return Regions.DEFAULT_REGION.getName();
+            } else if (values.length > 3) {
+                return values[1];
+            }
+        }
+
+        return Regions.DEFAULT_REGION.getName();
+    }
+
+    /**
+     * Retrieves the endpoint from a given URI.
+     * @param uri The URI to handle
+     * @return The endpoint
+     */
+    private String getEndPointFromUri(URI uri) {
+        String host = uri.getHost();
+        if (host.contains("amazonaws.com")) {
+            return ENDPOINT;
```

Comment:
Why is this falling back to the region unspecific endpoint?
According to https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/java-dg-region-selection.html#region-selection-choose-endpoint, the expectation is still for   `EndpointConfiguration` to receive a region-specific endpoint as its first argument.